modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/listk.kt

```diff
@@ -323,3 +325,11 @@ interface ListKSemialign : Semialign<ForListK>, ListKFunctor {
 interface ListKAlign : Align<ForListK>, ListKSemialign {
   override fun <A> empty(): Kind<ForListK, A> = ListK.empty()
 }
+
+@extension
+interface ListKUnalign : Unalign<ForListK>, ListKSemialign {
+  override fun <A, B> unalign(ior: Kind<ForListK, Ior<A, B>>): Tuple2<Kind<ForListK, A>, Kind<ForListK, B>> =
+    ior.fix().let { list ->
+      list.filterMap { it.toLeftOption() } toT list.filterMap { it.toOption() }
+    }
```

Comment:
Can you do this and the other ones in a single pass? Like: (pseudocode)
```kotlin
iorList.fold(emptyList() toT emptyList()) { (l, r), x ->
  x.fold(
    { l + it toT r },
    { l toT r + it },
    { a, b -> l + a toT r + b }
  )
}
```
Something like this might be better. All other container types can be done like this as well.

Reply:
true, i changed it for ListK and MapK. also added the missing SortedMapK unalign instance with such an implementation